packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,170 @@
+import inquirer from 'inquirer';
+import path from 'path';
+import fs from 'fs-extra';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+inquirer.registerPrompt('datetime', require('inquirer-datepicker'));
+import { CronBuilder } from '../utils/cronBuilder';
+import { constructCloudWatchEventComponent } from '../utils/cloudformationHelpers';
+import { minuteHelper, hourHelper, timeHelper, weekHelper, monthHelper, yearHelper } from '../utils/cronHelper';
+import { MINUTES, HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY, CUSTOM } from '../utils/constants';
+import { CronExpression } from '../utils/cronExpression';
+const categoryName = 'function';
+
+export async function scheduleWalkthrough(context: any, params: Partial<FunctionParameters>): Promise<Partial<FunctionParameters>> {
```

Comment:
would be better if the return value was `Promise<Pick<FunctionParameters, 'cloudwatchRule'>>` so it was obvious from the method signature exactly what parameters it was supplying